packages/profile-sync-controller/src/controllers/user-storage/network-syncing/sync-all.ts

```diff
@@ -0,0 +1,238 @@
+  const localNetworksToUpdate: NetworkConfiguration[] = [];
+  const localNetworksToRemove: NetworkConfiguration[] = [];
+
+  // Get networks in both, these need to be checked
+  const networksInBoth = setIntersection(localKeySet, remoteKeySet);
+  networksInBoth.forEach((chain) => {
+    const localNetwork = localMap.get(chain);
+    const remoteNetwork = remoteMap.get(chain);
+    if (!localNetwork || !remoteNetwork) {
+      return;
+    }
+
+    const whichIsLatest = checkWhichNetworkIsLatest(
+      localNetwork,
+      remoteNetwork,
+    );
+
+    // Local Wins -> Need to update remote
+    if (whichIsLatest === 'Local Wins') {
+      remoteNetworksToUpdate.push(toRemoteNetworkConfiguration(localNetwork));
+    }
+
+    // Remote Wins (and isn't deleted) -> Need to update local
+    if (whichIsLatest === 'Remote Wins') {
+      if (remoteNetwork.d) {
+        localNetworksToRemove.push(toNetworkConfiguration(remoteNetwork));
+      } else {
+        localNetworksToUpdate.push(toNetworkConfiguration(remoteNetwork));
+      }
+    }
+  });
+
+  return {
+    remoteNetworksToUpdate,
+    localNetworksToUpdate,
+    localNetworksToRemove,
+  };
+};
+
+export const getNewLocalNetworks = (props: {
+  originalList: NetworkConfiguration[];
+  missingLocalNetworks: NetworkConfiguration[];
+  localNetworksToUpdate: NetworkConfiguration[];
+  localNetworksToRemove: NetworkConfiguration[];
+}) => {
+  let newList = [...props.originalList];
+  newList.push(...props.missingLocalNetworks);
+  const updateMap = createMap(props.localNetworksToUpdate);
+  const remoteMap = createMap(props.localNetworksToRemove);
+
+  newList = newList
+    .map((n) => {
+      if (remoteMap.has(n.chainId)) {
+        return undefined;
+      }
+
+      const updatedNetwork = updateMap.get(n.chainId);
+      if (updatedNetwork) {
+        return updatedNetwork;
+      }
+
+      return n;
+    })
+    .filter((n): n is NetworkConfiguration => n !== undefined);
+
+  return newList;
+};
+
+export const findNetworksToUpdate = (props: FindNetworksToUpdateProps) => {
+  try {
+    const { localNetworks, remoteNetworks } = props;
+
+    // Get Maps & Key Sets
+    const ds = getDataStructures(localNetworks, remoteNetworks);
+
+    // Calc Missing Networks
+    const missingNetworks = getMissingNetworkLists(ds);
+
+    // Calc Updated Networks
+    const updatedNetworks = getUpdatedNetworkLists(ds);
+
+    // List of networks we need to update
+    const remoteNetworksToUpdate = [
+      ...missingNetworks.missingRemoteNetworks,
+      ...updatedNetworks.remoteNetworksToUpdate,
+    ];
+
+    // List of new local networks
+    const newLocalNetworks = getNewLocalNetworks({
+      originalList: localNetworks,
+      missingLocalNetworks: missingNetworks.missingLocalNetworks,
+      localNetworksToRemove: updatedNetworks.localNetworksToRemove,
+      localNetworksToUpdate: updatedNetworks.localNetworksToUpdate,
+    });
```

Comment:
I'm not entirely convinced we need this combined local network lists.

Depending on the integration, we may choose to swap this for the separated lists (might make it easier to determine which lists need updating or removing)

Reply:
That is fine, we can change this in a future PR depending if the integration with the external `NetworkController` interface.